altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -23,100 +55,204 @@ impl UsartControl {
         }
     }
 
-    pub fn enable_usart(&self) {
+    // --------------------------------------------------------------
+    /// This section pertains to calls made to control register one (cr1).
+
+    /// Makes call to control register one, passing in the value 'true' in
+    /// order to enable the Usart.
+    pub fn enable_usart(&mut self) {
         self.cr1.enable_usart(true);
     }
 
-    pub fn disable_usart(&self) {
+    /// Makes call to control register one, passing in the value 'false' in
+    /// order to disable the Usart.
+    pub fn disable_usart(&mut self) {
         self.cr1.enable_usart(false);
     }
 
+    /// Makes call to control register one to check if usart is enabled or not.
+    /// Returns true if on. False otherwise.
     pub fn is_usart_enabled(&self) -> bool {
         self.cr1.is_usart_enabled()
     }
 
-    pub fn set_word_length(&self, length: WordLength) {
-        self.cr1.set_word_length(length);
+    /// Makes call to control register one to set the Usart mode.
+    /// Possible Mode settings are: None, Receive, Transmit, All
+    pub fn set_mode(&mut self, mode: Mode) {
+        self.cr1.set_mode(mode);
     }
 
-    pub fn set_mode(&self, mode: Mode) {
-        self.cr1.set_mode(mode);
+    /// Makes call to control register one to enable the RXNE interrupt.
+    pub fn enable_receiver_not_empty_interrupt(&mut self) {
+        self.cr1.set_receiver_not_empty_interrupt(true);
+    }
+
+    /// Makes call to control register one to disable the RXNE interrupt.
+    pub fn disable_receiver_not_empty_interrupt(&mut self) {
+        self.cr1.set_receiver_not_empty_interrupt(false);
+    }
+
+    /// Makes call to control register one to enable the TC interrupt.
+    pub fn enable_transmit_complete_interrupt(&mut self) {
+        self.cr1.set_transmit_complete_interrupt(true);
+    }
+
+    /// Makes call to control register one to disable the TC interrupt.
+    pub fn disable_transmit_complete_interrupt(&mut self) {
+        self.cr1.set_transmit_complete_interrupt(false);
+    }
+
+    /// Makes call to control register one to enable the TXEIE interrupt.
+    pub fn enable_transmit_interrupt(&mut self) {
+        self.cr1.set_transmit_interrupt(true);
+    }
+
+    /// Makes call to control register one to disable the TXEIE interrupt.
+    pub fn disable_transmit_interrupt(&mut self) {
+        self.cr1.set_transmit_interrupt(false);
     }
 
-    pub fn set_parity(&self, parity: Parity) {
+    /// Makes call to control register one to set the parity.
+    /// Possible Parity settings are: None, Even, Odd
+    pub fn set_parity(&mut self, parity: Parity) {
         self.cr1.set_parity(parity);
     }
 
-    pub fn set_stop_bits(&self, length: StopLength) {
-        self.cr2.set_stop_bits(length);
+    /// Makes call to control register one to set the word length.
```

Comment:
You could explain what setting the word length means, the bit length of each data packet right?